packages/block-library/src/navigation/index.php

```diff
@@ -132,6 +132,83 @@ function block_core_navigation_render_submenu_icon() {
 	return '<svg xmlns="http://www.w3.org/2000/svg" width="12" height="12" viewBox="0 0 12 12" fill="none" role="img" aria-hidden="true" focusable="false"><path d="M1.50002 4L6.00002 8L10.5 4" stroke-width="1.5"></path></svg>';
 }
 
+
+/**
+ * Finds the first non-empty `wp_navigation` Post.
+ *
+ * @return WP_Post|null the first non-empty Navigation or null.
+ */
+function block_core_navigation_get_non_empty_navigation() {
+	// Order and orderby args set to mirror those in `wp_get_nav_menus`
+	// see:
+	// - https://github.com/WordPress/wordpress-develop/blob/ba943e113d3b31b121f77a2d30aebe14b047c69d/src/wp-includes/nav-menu.php#L613-L619.
+	// - https://developer.wordpress.org/reference/classes/wp_query/#order-orderby-parameters.
+	$navigation_posts = get_posts(
+		array(
+			'post_type'      => 'wp_navigation',
+			'order'          => 'ASC',
+			'orderby'        => 'name',
+			'posts_per_page' => -1, // include all posts.
```

Comment:
```suggestion
			'posts_per_page' => 1, // only the first post
```

Reply:
As @noisysocks noticed, there may be more posts than RAM. This function returns only the first one anyway, so it doesn't have to process the other ones.

Reply:
Ah right I see. So we're querying first the first Nav Post which has block grammar inside it. Makes sense. At first I assumed it was a typo!